markdownlint/TOP010_useLazyNumbering/TOP010_useLazyNumbering.js

```diff
@@ -0,0 +1,37 @@
+module.exports = {
+  names: ["TOP010", "ordered-list-for-lazy-numbering"],
+  description: "Ordered lists should use lazy numbering",
+  information: new URL(
+    "https://github.com/TheOdinProject/curriculum/blob/main/markdownlint/docs/TOP010.md"
+  ),
+  tags: ["lazy-numbering"],
+  parser: "markdownit",
+  function: function TOP010(params, onError) {
+    params.parsers.markdownit.tokens.forEach((token) => {
+      if (
+        token.tag === "li" &&
+        token.line !== undefined &&
+        /\d/.test(token.line) &&
+        !/^\s*1\./.test(token.line)
+      ) {
+        const originalText = token.line;
+        // https://regexr.com/80oan to test this regex
+        const updatedText = originalText.replace(/^\s*\d+\./, (match) =>
+          match.replace(/\d+/, "1")
+        );
+        const textLength = token.line.length;
+        const lineNumber = token.lineNumber;
+
+        onError({
+          lineNumber: lineNumber,
+          detail: `\n  Expected: "${updatedText}"\n  Actual: "${originalText}"\n`,
```

Comment:
Ideally, we should only report `Expected: "1."; Actual: "2."` or similar, since we're only concerned with the ol prefix.
If we include the entire line, the following can realistically occur, which is a lot of noise:
![image](https://github.com/TheOdinProject/curriculum/assets/122839503/85ae0169-37f2-4326-a8a3-9d461e87bdee)

Reply:
This is a good point! I didn't even consider this! I will fix!